app/src/main/java/org/fcitx/fcitx5/android/input/FcitxInputMethodService.kt

```diff
@@ -111,6 +113,12 @@ class FcitxInputMethodService : LifecycleInputMethodService() {
             fcitx.runImmediately { eventFlow.onEach(::handleFcitxEvent).launchIn(lifecycleScope) }
         AppPrefs.getInstance().apply {
             keyboard.buttonHapticFeedback.registerOnChangeListener(recreateInputViewListener)
+            keyboard.buttonPressVibrationMilliseconds.registerOnChangeListener(recreateInputViewListener)
+            keyboard.buttonLongPressVibrationMilliseconds.registerOnChangeListener(recreateInputViewListener)
+            if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.O && vibrator.hasAmplitudeControl()){
+                keyboard.buttonPressVibrationAmplitude.registerOnChangeListener(recreateInputViewListener)
+                keyboard.buttonLongPressVibrationMilliseconds.registerOnChangeListener(recreateInputViewListener)
+            }
```

Comment:
no need to `registerOnChangeListener` here, because it's value is read every time when `CustomGestureView#hapticFeedback(int)` is called